spec-tests/src/test/java/org/candlepin/spec/jobs/InactiveConsumerCleanerJobTest.java

```diff
@@ -0,0 +1,171 @@
+        jobsApi = client.jobs();
+
+        if (owner == null) {
+            owner = ownerApi.createOwner(Owners.random());
+        }
+    }
+
+    @Test
+    @DisplayName("should delete inactive consumers")
+    public void shouldDeleteInactiveConsumers() throws Exception {
+        Instant inactiveTime = Instant.now()
+            .minus(DEFAULT_LAST_CHECKED_IN_RETENTION_IN_DAYS + 10, ChronoUnit.DAYS);
+        Instant activeTime = Instant.now()
+            .minus(DEFAULT_LAST_CHECKED_IN_RETENTION_IN_DAYS - 10, ChronoUnit.DAYS);
+
+        ConsumerDTO inactiveConsumer = new Consumers.Builder()
+            .withOwner(owner)
+            .withLastCheckin(inactiveTime.atOffset(ZoneOffset.UTC))
+            .build();
+
+        ConsumerTypeDTO manifestType = new ConsumerTypeDTO();
+        manifestType.setManifest(true);
+        manifestType.setLabel("candlepin");
+        ConsumerDTO consumerWithManifest = new Consumers.Builder()
+            .withOwner(owner)
+            .withType(manifestType)
+            .withLastCheckin(inactiveTime.atOffset(ZoneOffset.UTC))
+            .build();
+        ConsumerDTO consumerWithEntitlement = new Consumers.Builder()
+            .withOwner(owner)
+            .withLastCheckin(inactiveTime.atOffset(ZoneOffset.UTC))
+            .build();
+        ConsumerDTO activeConsumer = new Consumers.Builder()
+            .withOwner(owner)
+            .withLastCheckin(activeTime.atOffset(ZoneOffset.UTC))
+            .build();
+        ConsumerDTO activeConsumer2 = new Consumers.Builder()
+            .withOwner(owner)
+            .withLastCheckin(activeTime.atOffset(ZoneOffset.UTC))
+            .build();
+
+        inactiveConsumer = consumerApi.createConsumer(inactiveConsumer, null, owner.getKey(), null, false);
+        consumerWithManifest = consumerApi
+            .createConsumer(consumerWithManifest, null, owner.getKey(), null, false);
+        consumerWithEntitlement = consumerApi
+            .createConsumer(consumerWithEntitlement, null, owner.getKey(), null, false);
+        activeConsumer = consumerApi.createConsumer(activeConsumer, null, owner.getKey(), null, false);
+        activeConsumer2 = consumerApi.createConsumer(activeConsumer2, null, owner.getKey(), null, false);
+
+        createProductAndPoolForConsumer(consumerWithEntitlement);
+
+        String jobId = jobsApi.scheduleJob(JOB_KEY).getId();
+        AsyncJobStatusDTO status = JobUtil.waitForJobToComplete(jobId, jobsApi);
+        assertEquals("FINISHED", status.getState());
+
+        // Verify that the inactive consumer has been deleted.
+        final String inactiveConsumer1Uuid = inactiveConsumer.getUuid();
+        assertGone(() -> consumerApi.getConsumer(inactiveConsumer1Uuid));
+        // Verify that the inactive consumer has been moved to the deleted_consumers table.
+        List<DeletedConsumerDTO> deletedConsumers = consumerApi
+            .listByDate(inactiveConsumer.getCreated().toString());
```

Comment:
I was very confused when I saw `consumerApi.listByDate` returning a list of `DeletedConsumerDTO`. It looks like the /deleted_consumers API gets bundled within the ConsumerApi class on the client side, which is very unintuitive/misleading. After a bit of digging, I found out the decision to roll APIs into the same client class happens based on the tags field we set on the spec yaml, and we happened to [set it to 'Consumer'](https://github.com/candlepin/candlepin/blob/5fa273be2319bf61008162428cef9e661c480d62/api/candlepin-api-spec.yaml#L2841-L2845) (along with all the /consumer endpoints). I just tested it and if we change the tag to `DeletedConsumer`, then we'll get this API rolled into it's own `DeletedConsumerApi` client class